### PR TITLE
add change lang tab for levels in gd or has assessment

### PR DESCRIPTION
**Pull request by smallst**
fix ENG-273 , can tested in gd levels for home users
image

### CODE REVIEW

File: app/locale/en.js [hunk 1]
```diff
@@ -1225,6 +1225,8 @@ module.exports = {
       auth_tab: 'Sign Up',
       inventory_caption: 'Equip your hero',
       choose_hero_caption: 'Choose hero, language',
+      change_langugae_caption: 'Choose language',
+      change_language_tab: 'Code Lang',
```
Comment (nwinter) [inline app/locale/en.js:1229]:
```suggestion
      change_language_tab: 'Code Language',
```

File: app/templates/play/menu/change-language-view.pug [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.form
+  .form-group.select-group.half-width.code-language-form
+    span.help-block(data-i18n="choose_hero.programming_language")
+    select#option-code-language(name="code-language")
+      for option in codeLanguages
+        option(value=option.id, selected=codeLanguage === option.id)= option.name
+  img.hr(src="/images/pages/play/modal/hr.png", draggable="false")
```
Comment (nwinter) [inline app/templates/play/menu/change-language-view.pug:7]:
```suggestion
```

File: app/views/play/menu/ChangeLanguageView.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+          this.codeLanguageList = _.filter(this.codeLanguageList, language => language.id !== 'coffeescript')
+        }
+      }
+    }
+
+    buildCodeFormats () {
+      const $select = this.$el.find('#option-code-format')
+      if (!$.browser.mobile) {
+        $select.fancySelect()
+      }
```
Comment (nwinter) [inline app/views/play/menu/ChangeLanguageView.js:126]: Test to make sure it works on mobile (this code should already do it)

File: app/views/play/menu/ChangeLanguageView.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    onCodeLanguageChanged (e) {
+      this.codeLanguage = this.$el.find('#option-code-language').val()
+      this.codeLanguageChanged = true
+      window.tracker?.trackEvent('Campaign changed code language', { category: 'Campaign Hero Select', codeLanguage: this.codeLanguage, levelSlug: this.options.level?.get('slug') })
+      if (this.codeFormat === 'blocks-and-code' && ['python', 'javascript'].indexOf(this.codeLanguage) === -1) {
+        // Blockly can't support languages like C++/Java. (Some day we'll have Lua.)
+        noty({ text: `Can't show blocks and code with ${this.codeLanguage}`, layout: 'bottomCenter', type: 'error', killer: false, timeout: 3000 })
+        this.$el.find('#option-code-format').val('text-code').change()
+      }
+      console.log(this.codeLanguage)
```
Comment (nwinter) [inline app/views/play/menu/ChangeLanguageView.js:161]:
```suggestion
```

File: app/views/play/menu/ChangeLanguageView.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    onCodeFormatChanged (e) {
+      this.codeFormat = this.$el.find('#option-code-format').val()
+      this.codeFormatChanged = true
+      window.tracker?.trackEvent('Campaign changed code format', { category: 'Campaign Hero Select', codeFormat: this.codeFormat, levelSlug: this.options.level?.get('slug') })
+      if (this.codeFormat === 'blocks-and-code' && ['python', 'javascript'].indexOf(this.codeLanguage) === -1) {
+        // Blockly can't support languages like C++/Java. (Some day we'll have Lua.)
+        noty({ text: `Can't show blocks and code with ${this.codeLanguage}`, layout: 'bottomCenter', type: 'error', killer: false, timeout: 3000 })
+        this.$el.find('#option-code-language').val('javascript').change()
+      }
+      console.log(this.codeFormat)
```
Comment (nwinter) [inline app/views/play/menu/ChangeLanguageView.js:173]:
```suggestion
```

File: app/views/play/menu/ChangeLanguageView.js
Comment (nwinter) [inline app/views/play/menu/ChangeLanguageView.js:1]: Ideally we refactor all this change language stuff out to a common component to DRY it from the hero modal

File: app/views/play/menu/ChangeLanguageView.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        me.set('aceConfig', aceConfig)
+        changed = true
+        codeFormatChanged = true
+      }
+
+      if (changed) { me.patch() }
+
+      // todo: what if change code-format?
+      if (codeFormatChanged) {
+        document.location.reload()
```
Comment (nwinter) [inline app/views/play/menu/ChangeLanguageView.js:81]: We have to wait for this to save before we reload, right?

**Review by nwinter — COMMENTED** [review]
Just a quick read so far

File: app/locale/en.js (hunk 1 above)
Comment (smallst) [reply app/locale/en.js:1229]: we can't. there's no space for full language text 😂

Comment (nwinter) [reply app/locale/en.js:1229]: Wrap it onto two lines or just say language?

**Pull request merged**